airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py

```diff
@@ -0,0 +1,353 @@
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, **kwargs)
+        params["owner"] = stream_slice.get("reference_id")
+        params["q"] = self.search_param
+        return params
+
+
+class AnalyticsStreamMixin(IncrementalLinkedinAdsStream):
+    """
+    AdAnalytics Streams more info:
+    https://docs.microsoft.com/en-us/linkedin/marketing/integrations/ads-reporting/ads-reporting?tabs=curl#ad-analytics
+    """
+
+    endpoint = "adAnalyticsV2"
+    # For Analytics streams the primary_key is the entity of the pivot [Campaign URN, Creative URN, etc] + `end_date`
+    primary_key = ["pivotValue", "end_date"]
+    cursor_field = "end_date"
+
+    @property
+    def base_analytics_params(self) -> MutableMapping[str, Any]:
+        """ Define the base parameters for analytics streams """
+        return {"q": "analytics", "pivot": self.pivot_by, "timeGranularity": "DAILY"}
+
+    def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = self.base_analytics_params
+        params[self.search_param] = f"{self.search_param_value}{stream_slice.get(self.primary_slice_key)}"
+        params.update(**update_analytics_params(stream_slice))
+        return params
+
+    def read_records(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {self.cursor_field: self.start_date}
+        parent_stream = self.slice_from_stream(config=self.config)
+        result_chunks = []
+        for record in parent_stream.read_records(sync_mode=SyncMode.full_refresh):
+            for analytics_slice in make_analytics_slices(record, self.slice_key_value_map, stream_state.get(self.cursor_field)):
+                child_stream_slice = super().read_records(stream_slice=analytics_slice, **kwargs)
+                result_chunks.append(child_stream_slice)
+            yield from merge_chunks(result_chunks, self.cursor_field)
+
+
+class AdCampaignAnalytics(AnalyticsStreamMixin):
+    """
+    Campaing Analytics stream.
+    See the AnalyticsStreamMixin class for more information.
+    """
+
+    slice_from_stream = Campaigns
+    slice_key_value_map = {"campaign_id": "id"}
+    search_param = "campaigns[0]"
+    search_param_value = "urn:li:sponsoredCampaign:"
+    pivot_by = "CAMPAIGN"
+
+
+class AdCreativeAnalytics(AnalyticsStreamMixin):
+    """
+    Creative Analytics stream.
+    See the AnalyticsStreamMixin class for more information.
+    """
+
+    slice_from_stream = Creatives
+    slice_key_value_map = {"creative_id": "id"}
+    search_param = "creatives[0]"
+    search_param_value = "urn:li:sponsoredCreative:"
+    pivot_by = "CREATIVE"
+
+
+class SourceLinkedinAds(AbstractSource):
```

Comment:
Could you add docs that describe what source Linkedin Ads do?

Reply:
Sure, good comment.